### PR TITLE
:book: Proposal for initial security policy

### PR DESCRIPTION
**Pull request by embik**
## Summary

Since we're working on moving the project towards maturity (as per the last few community meetings), we should establish a security policy to replace the "WIP" document that we have.

We can use GitHub's built-in security advisory process (which gives us goodies like GitHub assigning CVEs for us) for accepting security reports.

This is a draft/proposal for an initial security policy. I assume that we want to refine it at some point, but having one over having none is probably a good idea.

Please leave some feedback on this PR.

## Related issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by embik** [comment]
/hold

since we should have maintainer consensus and discussion of this during a community meeting before merging it.

**Comment by sttts** [comment]
Lgtm.

**Comment by sttts** [comment]
Pinging the other maintainers to give feedback: @clubanderson @scheeles @xrstf @mjudeikis

**Comment by mjudeikis** [comment]
lgtm apart of nit.

File: SECURITY.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+
+kcp uses GitHub to allow submission of private security reports. Please report any security finding via [this link](https://github.com/kcp-dev/kcp/security/advisories/new). Maintainers will triage your report as soon as possible and get in touch with you via your report in case they have more questions.
+
+As a security researcher, please report vulnerabilities to kcp in a [coordinated vulnerability disclosure](https://cheatsheetseries.owasp.org/cheatsheets/Vulnerability_Disclosure_Cheat_Sheet.html) fashion. In return, maintainers pledge to engage in good faith and collaborate with security researchers to address and publish vulnerabilities found in kcp.
+
+Please understand that the maintainers also do not accept results of dependency scanners without proof that the detected CVE / vulnerability can be used against kcp.
+
+## Security Advisories
+
+Public disclosure of vulnerabilities happens through GitHub. Please visit [Security Advisories](https://github.com/kcp-dev/kcp/security/advisories) to review security bulletins published by the maintainers.
```
[the remark this answers is not in the document]
Comment (embik) [reply SECURITY.md]: Added that we will use kcp-users to inform people as well.

**Comment by mjudeikis** [comment]
/lgtm
/approve

as we addressed the comments from the community call, let's merge it and iterate.

**Pull request merged**